kod/object/item.kod

```diff
@@ -615,11 +615,12 @@ messages:
       return 1;
    }
 
-   GetRarity()
+   GetRarity(bShow_all = FALSE)
```

Comment:
The "show_all" behavior isn't really related to the rarity of the item, so this check feels like it belongs in the caller, not in Item.

This parameter is also undocumented, which is a no-no, but no need to document it if you're removing it of course.

Reply:
This issue is confusing me.  Maybe I'm thinking about it wrong, but I don't see how I can intercept the rarity or modify it without sending a parameter to `GetRarity`.

When a player casts identify, it will go `CastSpell` in `identify.kod`
`Send(who,@SendLook,#what=oTarget,#bShow_All=TRUE);`

`SendLook `in turn will call `ToCliObject`
`Send(self,@ToCliObject,#what=what,#show_type=SHOW_LOOK,#bShow_All=bShow_All);`

Then I have my call to `GetRarity`
`rarity = Send(what,@GetRarity);`

I figured I could do a check to jump past the `IsIdentified `check in `GetRarity `if the original parameter of `bShow_All = TRUE`.  

I could create some kind of other function that doesn't have the unidentified check but that feels wrong and steps on the duplication of code issue.

I still need to run through all my checks to see what the item is (CURSED, MAGIC, RARE, LEGENDARY, NORMAL) and I can't think of a better place than the `GetRarity `function. I must skip the Identified section somehow.

I guess my question is: what would be the caller in your comment?  Is that `ToCliObject `that calls `GetRarity`? Or is that the original caller `CastSpell` in `identify.kod`?

Reply:
The caller of GetRarity is ToCliObject.  I think it makes more logical sense to do the "show all" check there, even though it's already a super enormous function.  Passing it as an argument to GetRarity muddles the behavior of that function.